### PR TITLE
Build wheels for Python 3.11

### PR DESCRIPTION
**Pull request by arcondello**
Also switch to manylinux2014.

### CODE REVIEW

**Comment by arcondello** [comment]
https://github.com/python/cpython/pull/29560 introduced a warning when a `SpooledTemporaryFile` is deleted without being previously closed. Going to need to make a lot of changes.

**Comment by arcondello** [comment]
Also, locally running unittests have gone from ~18s to ~60s. Will need to look into the source of the slowdown.

**Review by randomir — APPROVED** [review]
LGTM!

File: .circleci/config.yml [hunk 1]
```diff
@@ -450,7 +431,7 @@ workflows:
             - build-windows
           matrix:
             parameters:
-              python-version: *python-versions
+              python-version: [3.7.9, 3.8.9, 3.9.4, 3.10.0] # no 3.11 with nuget yet
```
[the remark this answers is not in the document]
Comment (arcondello) [reply .circleci/config.yml:434]: Great idea! Will look into it.

[the remark this answers is not in the document]
Comment (arcondello) [reply .circleci/config.yml]: Ok, this has turned into a hellscape of windows syntax. Specifically flavours of [this](https://forums.docker.com/t/cant-set-docker-volume-for-container-in-windows-docker-ce/31841). I have run out of patience so I think we should just skip 3.11 tests for now.

[the remark this answers is not in the document]
Comment (randomir) [reply .circleci/config.yml:434]: You can try with `shell: bash -eo pipefail`.

Or yes, just skip 3.11 tests for now.

[the remark this answers is not in the document]
Comment (arcondello) [reply .circleci/config.yml]: Ok, I have it working. Now the issue is that it takes 4+min to download the docker image! I tried seeing if I can speed it up with `docker save`/`docker load` but those are also stupidly slow.

[the remark this answers is not in the document]
Comment (randomir) [reply .circleci/config.yml]: Yes, that's awful. :nauseated_face: :scream:

I can't understand why CircleCI doesn't cache those 2GB+ docker images. I bet GitHub Actions start Windows images in seconds :stuck_out_tongue_closed_eyes:

So, I retract my initial idea of using Windows images.

**Pull request merged**